### ISSUE TITLE
`%config` line magic errors in notebooks

### ISSUE THREAD

**Issue by mathause** [jupyterlab/jupyterlab#13473]
## Description

Calling the `%config` line magic errors.

## Reproduce

```python
%config InlineBackend.figure_format = "retina"
```

Calling the above line in a notebook raises an error (in jupyterlab and jupyter notebookl). It does not seem to raise in a `ipython` console, which is why I report it here - but of course this might also belong somewhere else.

Traceback:

<details>

```python-traceback

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [1], line 1
----> 1 get_ipython().run_line_magic('config', 'InlineBackend.figure_format = "retina"')

File ~/.conda/envs/jupyterlabtest/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2364, in InteractiveShell.run_line_magic(self, magic_name, line, _stack_depth)
   2362     kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2363 with self.builtin_trap:
-> 2364     result = fn(*args, **kwargs)
   2365 return result

File ~/.conda/envs/jupyterlabtest/lib/python3.11/site-packages/IPython/core/magics/config.py:163, in ConfigMagics.config(self, s)
    159 from traitlets.config.loader import Config
    160 # some IPython objects are Configurable, but do not yet have
    161 # any configurable traits.  Exclude them from the effects of
    162 # this magic, as their presence is just noise:
--> 163 configurables = sorted(set([ c for c in self.shell.configurables
    164                              if c.__class__.class_traits(config=True)
    165                              ]), key=lambda x: x.__class__.__name__)
    166 classnames = [ c.__class__.__name__ for c in configurables ]
    168 line = s.strip()

File ~/.conda/envs/jupyterlabtest/lib/python3.11/site-packages/IPython/core/magics/config.py:164, in <listcomp>(.0)
    159 from traitlets.config.loader import Config
    160 # some IPython objects are Configurable, but do not yet have
    161 # any configurable traits.  Exclude them from the effects of
    162 # this magic, as their presence is just noise:
    163 configurables = sorted(set([ c for c in self.shell.configurables
--> 164                              if c.__class__.class_traits(config=True)
    165                              ]), key=lambda x: x.__class__.__name__)
    166 classnames = [ c.__class__.__name__ for c in configurables ]
    168 line = s.strip()

AttributeError: type object 'CommManager' has no attribute 'class_traits'
```
</details>

## Expected behavior

No error

## Context

- Operating System and version: 
- Browser and version: 
- JupyterLab version: 

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
        /home/mathause/.conda/envs/jupyterlabtest/bin
        /home/mathause/.conda/condabin
        /home/mathause/bin
        /usr/local/sbin
        /usr/local/bin
        /usr/sbin
        /usr/bin
        /sbin
        /bin
        /usr/games
        /usr/local/games
        /snap/bin

sys.path:
        /home/mathause/.conda/envs/jupyterlabtest/bin
        /home/mathause/.conda/envs/jupyterlabtest/lib/python311.zip
        /home/mathause/.conda/envs/jupyterlabtest/lib/python3.11
        /home/mathause/.conda/envs/jupyterlabtest/lib/python3.11/lib-dynload
        /home/mathause/.conda/envs/jupyterlabtest/lib/python3.11/site-packages

sys.executable:
        /home/mathause/.conda/envs/jupyterlabtest/bin/python

sys.version:
        3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:24:40) [GCC 10.4.0]

platform.platform():
        Linux-5.15.0-53-generic-x86_64-with-glibc2.31

which -a jupyter:
        /home/mathause/.conda/envs/jupyterlabtest/bin/jupyter

pip list:
        Package                       Version
        ----------------------------- -----------
        anyio                         3.6.2
        argon2-cffi                   21.3.0
        argon2-cffi-bindings          21.2.0
        asttokens                     2.1.0
        attrs                         22.1.0
        Babel                         2.11.0
        backcall                      0.2.0
        backports.functools-lru-cache 1.6.4
        beautifulsoup4                4.11.1
        bleach                        5.0.1
        brotlipy                      0.7.0
        certifi                       2022.9.24
        cffi                          1.15.1
        charset-normalizer            2.1.1
        comm                          0.1.0
        cryptography                  38.0.3
        debugpy                       1.6.3
        decorator                     5.1.1
        defusedxml                    0.7.1
        entrypoints                   0.4
        executing                     1.2.0
        fastjsonschema                2.16.2
        flit_core                     3.8.0
        idna                          3.4
        importlib-metadata            5.0.0
        importlib-resources           5.10.0
        ipykernel                     6.18.0
        ipython                       8.6.0
        ipython-genutils              0.2.0
        jedi                          0.18.2
        Jinja2                        3.1.2
        json5                         0.9.5
        jsonschema                    4.17.1
        jupyter_client                7.4.7
        jupyter_core                  5.0.0
        jupyter-server                1.23.3
        jupyterlab                    3.5.0
        jupyterlab-pygments           0.2.2
        jupyterlab_server             2.16.3
        MarkupSafe                    2.1.1
        matplotlib-inline             0.1.6
        mistune                       2.0.4
        nbclassic                     0.4.8
        nbclient                      0.7.0
        nbconvert                     7.2.5
        nbformat                      5.7.0
        nest-asyncio                  1.5.6
        notebook                      6.5.2
        notebook_shim                 0.2.2
        packaging                     21.3
        pandocfilters                 1.5.0
        parso                         0.8.3
        pexpect                       4.8.0
        pickleshare                   0.7.5
        pip                           22.3.1
        pkgutil_resolve_name          1.3.10
        platformdirs                  2.5.2
        prometheus-client             0.15.0
        prompt-toolkit                3.0.33
        psutil                        5.9.4
        ptyprocess                    0.7.0
        pure-eval                     0.2.2
        pycparser                     2.21
        Pygments                      2.13.0
        pyOpenSSL                     22.1.0
        pyparsing                     3.0.9
        pyrsistent                    0.19.2
        PySocks                       1.7.1
        python-dateutil               2.8.2
        pytz                          2022.6
        pyzmq                         24.0.1
        requests                      2.28.1
        Send2Trash                    1.8.0
        setuptools                    65.5.1
        six                           1.16.0
        sniffio                       1.3.0
        soupsieve                     2.3.2.post1
        stack-data                    0.6.1
        terminado                     0.17.0
        tinycss2                      1.2.1
        tomli                         2.0.1
        tornado                       6.2
        traitlets                     5.5.0
        typing_extensions             4.4.0
        urllib3                       1.26.11
        wcwidth                       0.2.5
        webencodings                  0.5.1
        websocket-client              1.4.2
        wheel                         0.38.4
        zipp                          3.10.0

conda list:
        # packages in environment at /home/mathause/.conda/envs/jupyterlabtest:
        #
        # Name                    Version                   Build  Channel
        _libgcc_mutex             0.1                 conda_forge    conda-forge
        _openmp_mutex             4.5                       2_gnu    conda-forge
        anyio                     3.6.2              pyhd8ed1ab_0    conda-forge
        argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
        argon2-cffi-bindings      21.2.0          py311hd4cff14_3    conda-forge
        asttokens                 2.1.0              pyhd8ed1ab_0    conda-forge
        attrs                     22.1.0             pyh71513ae_1    conda-forge
        babel                     2.11.0             pyhd8ed1ab_0    conda-forge
        backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
        backports                 1.0                pyhd8ed1ab_3    conda-forge
        backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
        beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
        bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
        brotlipy                  0.7.0           py311hd4cff14_1005    conda-forge
        bzip2                     1.0.8                h7f98852_4    conda-forge
        ca-certificates           2022.9.24            ha878542_0    conda-forge
        certifi                   2022.9.24          pyhd8ed1ab_0    conda-forge
        cffi                      1.15.1          py311h409f033_2    conda-forge
        charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
        comm                      0.1.0              pyhd8ed1ab_0    conda-forge
        cryptography              38.0.3          py311h42a1071_0    conda-forge
        debugpy                   1.6.3           py311ha362b79_1    conda-forge
        decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
        defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
        entrypoints               0.4                pyhd8ed1ab_0    conda-forge
        executing                 1.2.0              pyhd8ed1ab_0    conda-forge
        flit-core                 3.8.0              pyhd8ed1ab_0    conda-forge
        idna                      3.4                pyhd8ed1ab_0    conda-forge
        importlib-metadata        5.0.0              pyha770c72_1    conda-forge
        importlib_resources       5.10.0             pyhd8ed1ab_0    conda-forge
        ipykernel                 6.18.0             pyh210e3f2_0    conda-forge
        ipython                   8.6.0              pyh41d4057_1    conda-forge
        ipython_genutils          0.2.0                      py_1    conda-forge
        jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
        jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
        json5                     0.9.5              pyh9f0ad1d_0    conda-forge
        jsonschema                4.17.1             pyhd8ed1ab_0    conda-forge
        jupyter_client            7.4.7              pyhd8ed1ab_0    conda-forge
        jupyter_core              5.0.0           py311h38be061_0    conda-forge
        jupyter_server            1.23.3             pyhd8ed1ab_0    conda-forge
        jupyterlab                3.5.0              pyhd8ed1ab_0    conda-forge
        jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
        jupyterlab_server         2.16.3             pyhd8ed1ab_0    conda-forge
        ld_impl_linux-64          2.39                 hcc3a1bd_1    conda-forge
        libffi                    3.4.2                h7f98852_5    conda-forge
        libgcc-ng                 12.2.0              h65d4601_19    conda-forge
        libgomp                   12.2.0              h65d4601_19    conda-forge
        libnsl                    2.0.0                h7f98852_0    conda-forge
        libsodium                 1.0.18               h36c2ea0_1    conda-forge
        libsqlite                 3.40.0               h753d276_0    conda-forge
        libstdcxx-ng              12.2.0              h46fd767_19    conda-forge
        libuuid                   2.32.1            h7f98852_1000    conda-forge
        libzlib                   1.2.13               h166bdaf_4    conda-forge
        markupsafe                2.1.1           py311hd4cff14_2    conda-forge
        matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
        mistune                   2.0.4              pyhd8ed1ab_0    conda-forge
        nbclassic                 0.4.8              pyhd8ed1ab_0    conda-forge
        nbclient                  0.7.0              pyhd8ed1ab_0    conda-forge
        nbconvert                 7.2.5              pyhd8ed1ab_0    conda-forge
        nbconvert-core            7.2.5              pyhd8ed1ab_0    conda-forge
        nbconvert-pandoc          7.2.5              pyhd8ed1ab_0    conda-forge
        nbformat                  5.7.0              pyhd8ed1ab_0    conda-forge
        ncurses                   6.3                  h27087fc_1    conda-forge
        nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
        notebook                  6.5.2              pyha770c72_1    conda-forge
        notebook-shim             0.2.2              pyhd8ed1ab_0    conda-forge
        openssl                   3.0.7                h166bdaf_0    conda-forge
        packaging                 21.3               pyhd8ed1ab_0    conda-forge
        pandoc                    2.19.2               h32600fe_1    conda-forge
        pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
        parso                     0.8.3              pyhd8ed1ab_0    conda-forge
        pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
        pickleshare               0.7.5                   py_1003    conda-forge
        pip                       22.3.1             pyhd8ed1ab_0    conda-forge
        pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
        platformdirs              2.5.2              pyhd8ed1ab_1    conda-forge
        prometheus_client         0.15.0             pyhd8ed1ab_0    conda-forge
        prompt-toolkit            3.0.33             pyha770c72_0    conda-forge
        psutil                    5.9.4           py311hd4cff14_0    conda-forge
        ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
        pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
        pycparser                 2.21               pyhd8ed1ab_0    conda-forge
        pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
        pyopenssl                 22.1.0             pyhd8ed1ab_0    conda-forge
        pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
        pyrsistent                0.19.2          py311hd4cff14_0    conda-forge
        pysocks                   1.7.1              pyha2e5f31_6    conda-forge
        python                    3.11.0          ha86cf86_0_cpython    conda-forge
        python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
        python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
        python_abi                3.11                    3_cp311    conda-forge
        pytz                      2022.6             pyhd8ed1ab_0    conda-forge
        pyzmq                     24.0.1          py311ha4b6469_1    conda-forge
        readline                  8.1.2                h0f457ee_0    conda-forge
        requests                  2.28.1             pyhd8ed1ab_1    conda-forge
        send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
        setuptools                65.5.1             pyhd8ed1ab_0    conda-forge
        six                       1.16.0             pyh6c4a22f_0    conda-forge
        sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
        soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
        stack_data                0.6.1              pyhd8ed1ab_0    conda-forge
        terminado                 0.17.0             pyh41d4057_0    conda-forge
        tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
        tk                        8.6.12               h27826a3_0    conda-forge
        tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
        tornado                   6.2             py311hd4cff14_1    conda-forge
        traitlets                 5.5.0              pyhd8ed1ab_0    conda-forge
        typing_extensions         4.4.0              pyha770c72_0    conda-forge
        tzdata                    2022f                h191b570_0    conda-forge
        urllib3                   1.26.11            pyhd8ed1ab_0    conda-forge
        wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
        webencodings              0.5.1                      py_1    conda-forge
        websocket-client          1.4.2              pyhd8ed1ab_0    conda-forge
        wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
        xz                        5.2.6                h166bdaf_0    conda-forge
        zeromq                    4.3.4                h9c3ff4c_1    conda-forge
        zipp                      3.10.0             pyhd8ed1ab_0    conda-forge

conda env:
        name: jupyterlabtest
        channels:
          - conda-forge
          - defaults
        dependencies:
          - _libgcc_mutex=0.1=conda_forge
          - _openmp_mutex=4.5=2_gnu
          - anyio=3.6.2=pyhd8ed1ab_0
          - argon2-cffi=21.3.0=pyhd8ed1ab_0
          - argon2-cffi-bindings=21.2.0=py311hd4cff14_3
          - asttokens=2.1.0=pyhd8ed1ab_0
          - attrs=22.1.0=pyh71513ae_1
          - babel=2.11.0=pyhd8ed1ab_0
          - backcall=0.2.0=pyh9f0ad1d_0
          - backports=1.0=pyhd8ed1ab_3
          - backports.functools_lru_cache=1.6.4=pyhd8ed1ab_0
          - beautifulsoup4=4.11.1=pyha770c72_0
          - bleach=5.0.1=pyhd8ed1ab_0
          - brotlipy=0.7.0=py311hd4cff14_1005
          - bzip2=1.0.8=h7f98852_4
          - ca-certificates=2022.9.24=ha878542_0
          - certifi=2022.9.24=pyhd8ed1ab_0
          - cffi=1.15.1=py311h409f033_2
          - charset-normalizer=2.1.1=pyhd8ed1ab_0
          - comm=0.1.0=pyhd8ed1ab_0
          - cryptography=38.0.3=py311h42a1071_0
          - debugpy=1.6.3=py311ha362b79_1
          - decorator=5.1.1=pyhd8ed1ab_0
          - defusedxml=0.7.1=pyhd8ed1ab_0
          - entrypoints=0.4=pyhd8ed1ab_0
          - executing=1.2.0=pyhd8ed1ab_0
          - flit-core=3.8.0=pyhd8ed1ab_0
          - idna=3.4=pyhd8ed1ab_0
          - importlib-metadata=5.0.0=pyha770c72_1
          - importlib_resources=5.10.0=pyhd8ed1ab_0
          - ipykernel=6.18.0=pyh210e3f2_0
          - ipython=8.6.0=pyh41d4057_1
          - ipython_genutils=0.2.0=py_1
          - jedi=0.18.2=pyhd8ed1ab_0
          - jinja2=3.1.2=pyhd8ed1ab_1
          - json5=0.9.5=pyh9f0ad1d_0
          - jsonschema=4.17.1=pyhd8ed1ab_0
          - jupyter_client=7.4.7=pyhd8ed1ab_0
          - jupyter_core=5.0.0=py311h38be061_0
          - jupyter_server=1.23.3=pyhd8ed1ab_0
          - jupyterlab=3.5.0=pyhd8ed1ab_0
          - jupyterlab_pygments=0.2.2=pyhd8ed1ab_0
          - jupyterlab_server=2.16.3=pyhd8ed1ab_0
          - ld_impl_linux-64=2.39=hcc3a1bd_1
          - libffi=3.4.2=h7f98852_5
          - libgcc-ng=12.2.0=h65d4601_19
          - libgomp=12.2.0=h65d4601_19
          - libnsl=2.0.0=h7f98852_0
          - libsodium=1.0.18=h36c2ea0_1
          - libsqlite=3.40.0=h753d276_0
          - libstdcxx-ng=12.2.0=h46fd767_19
          - libuuid=2.32.1=h7f98852_1000
          - libzlib=1.2.13=h166bdaf_4
          - markupsafe=2.1.1=py311hd4cff14_2
          - matplotlib-inline=0.1.6=pyhd8ed1ab_0
          - mistune=2.0.4=pyhd8ed1ab_0
          - nbclassic=0.4.8=pyhd8ed1ab_0
          - nbclient=0.7.0=pyhd8ed1ab_0
          - nbconvert=7.2.5=pyhd8ed1ab_0
          - nbconvert-core=7.2.5=pyhd8ed1ab_0
          - nbconvert-pandoc=7.2.5=pyhd8ed1ab_0
          - nbformat=5.7.0=pyhd8ed1ab_0
          - ncurses=6.3=h27087fc_1
          - nest-asyncio=1.5.6=pyhd8ed1ab_0
          - notebook=6.5.2=pyha770c72_1
          - notebook-shim=0.2.2=pyhd8ed1ab_0
          - openssl=3.0.7=h166bdaf_0
          - packaging=21.3=pyhd8ed1ab_0
          - pandoc=2.19.2=h32600fe_1
          - pandocfilters=1.5.0=pyhd8ed1ab_0
          - parso=0.8.3=pyhd8ed1ab_0
          - pexpect=4.8.0=pyh1a96a4e_2
          - pickleshare=0.7.5=py_1003
          - pip=22.3.1=pyhd8ed1ab_0
          - pkgutil-resolve-name=1.3.10=pyhd8ed1ab_0
          - platformdirs=2.5.2=pyhd8ed1ab_1
          - prometheus_client=0.15.0=pyhd8ed1ab_0
          - prompt-toolkit=3.0.33=pyha770c72_0
          - psutil=5.9.4=py311hd4cff14_0
          - ptyprocess=0.7.0=pyhd3deb0d_0
          - pure_eval=0.2.2=pyhd8ed1ab_0
          - pycparser=2.21=pyhd8ed1ab_0
          - pygments=2.13.0=pyhd8ed1ab_0
          - pyopenssl=22.1.0=pyhd8ed1ab_0
          - pyparsing=3.0.9=pyhd8ed1ab_0
          - pyrsistent=0.19.2=py311hd4cff14_0
          - pysocks=1.7.1=pyha2e5f31_6
          - python=3.11.0=ha86cf86_0_cpython
          - python-dateutil=2.8.2=pyhd8ed1ab_0
          - python-fastjsonschema=2.16.2=pyhd8ed1ab_0
          - python_abi=3.11=3_cp311
          - pytz=2022.6=pyhd8ed1ab_0
          - pyzmq=24.0.1=py311ha4b6469_1
          - readline=8.1.2=h0f457ee_0
          - requests=2.28.1=pyhd8ed1ab_1
          - send2trash=1.8.0=pyhd8ed1ab_0
          - setuptools=65.5.1=pyhd8ed1ab_0
          - six=1.16.0=pyh6c4a22f_0
          - sniffio=1.3.0=pyhd8ed1ab_0
          - soupsieve=2.3.2.post1=pyhd8ed1ab_0
          - stack_data=0.6.1=pyhd8ed1ab_0
          - terminado=0.17.0=pyh41d4057_0
          - tinycss2=1.2.1=pyhd8ed1ab_0
          - tk=8.6.12=h27826a3_0
          - tomli=2.0.1=pyhd8ed1ab_0
          - tornado=6.2=py311hd4cff14_1
          - traitlets=5.5.0=pyhd8ed1ab_0
          - typing_extensions=4.4.0=pyha770c72_0
          - tzdata=2022f=h191b570_0
          - urllib3=1.26.11=pyhd8ed1ab_0
          - wcwidth=0.2.5=pyh9f0ad1d_2
          - webencodings=0.5.1=py_1
          - websocket-client=1.4.2=pyhd8ed1ab_0
          - wheel=0.38.4=pyhd8ed1ab_0
          - xz=5.2.6=h166bdaf_0
          - zeromq=4.3.4=h9c3ff4c_1
          - zipp=3.10.0=pyhd8ed1ab_0
        prefix: /home/mathause/.conda/envs/jupyterlabtest
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

**Comment by jweill-aws** [comment]
@krassowski points out that https://github.com/ipython/ipykernel/pull/1037 may have fixed this problem in ipykernel 6.18.2. The newest version of ipykernel is 6.18.3. Can you please try upgrading to 6.18.3 and seeing if this fixes your issue?

**Comment by mathause** [comment]
Thanks! ipykernel 6.18.3 does indeed fix the problem, as does downgrading to 6.17.1.
